### PR TITLE
adding rhel-system-repo support for the lab

### LINKED ISSUE #12: MacOS mojave: cask and vagrant package missing
Hello, 
I run the command to install the environment but it resulted with error:
...

See: https://docs.brew.sh/Homebrew-and-Python
Updating Homebrew...
Error: Unknown command: cask
Error: Unknown command: cask
Error: Unknown command: cask
-bash: vagrant: command not found

I tried to install cask and vagrant 

Could you help me ?

### LINKED ISSUE #13: Wrong clone information for rhce8
Install the Virtual Box Extension Pack

Once the above software is installed. Do the following if you're running the environment on Mac:

    Create a separate ~/bin directory and cd to it. (The directory doesn't have to be ~/bin, it can be anything you want.)
    Clone the environment repo to it with git clone https://github.com/rdbreak/rhcsa8env.git
    Change to the rhcsa8env directory that is now in your ~/bin directory.
    Run vagrant up to deploy the environment (If the environment has a designated repo VM it will take the longest to deploy the first time only, this is because the repo system has all the packages available to the base release but will be quicker on subsequent deployments.)

Need to update the rhcsa8env with rhce8env ****

### LINKED ISSUE #14: Repo vm will not provision with an error about not enough space for pip to install ansible.
Environments:  RHEL 8.4 and Fedora 33.
F33 running the following pkgs:

vagrant-libvirt-0.1.2-1.fc33.noarch
vagrant-2.2.9-3.fc33.noarch
virtualbox-guest-additions-6.1.18-1.fc33.x86_64
VirtualBox-kmodsrc-6.1.18-1.fc33.noarch
akmod-VirtualBox-6.1.18-1.fc33.x86_64
VirtualBox-server-6.1.18-1.fc33.x86_64
VirtualBox-6.1.18-1.fc33.x86_64
kmod-VirtualBox-5.11.8-200.fc33.x86_64-6.1.18-1.fc33.x86_64

Running 'vagrant up' results in:
```
==> repo: Running provisioner: shell...
    repo: Running: inline script
    repo: Updating Subscription Management repositories.
    repo: Unable to read consumer identity
    repo: 
    repo: This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.
    repo: Last metadata expiration check: 0:47:29 ago on Wed 24 Mar 2021 09:43:18 PM PDT.
    repo: epel-release-latest-7.noarch.rpm                 64 kB/s |  15 kB     00:00    
    repo: Package epel-release-7-13.noarch is already installed.
    repo: Dependencies resolved.
    repo: Nothing to do.
    repo: Complete!
    repo: Updating Subscription Management repositories.
    repo: Unable to read consumer identity
    repo: 
    repo: This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.
    repo: Last metadata expiration check: 0:47:30 ago on Wed 24 Mar 2021 09:43:18 PM PDT.
    repo: Package sshpass-1.06-1.el7.x86_64 is already installed.
    repo: Package python3-pip-9.0.3-13.el8.noarch is already installed.
    repo: Package python36-devel-3.6.8-1.module+el8+2710+846623d6.x86_64 is already installed.
    repo: Package httpd-2.4.37-30.module+el8.3.0+7001+0766b9e7.x86_64 is already installed.
    repo: Package vsftpd-3.0.3-28.el8.x86_64 is already installed.
    repo: Package createrepo_c-0.11.0-1.el8.x86_64 is already installed.
    repo: Dependencies resolved.
    repo: Nothing to do.
    repo: Complete!
==> repo: Running provisioner: shell...
    repo: Running: inline script
    repo: Requirement already satisfied: pip in /usr/local/lib/python3.6/site-packages (21.0.1)
    repo: Requirement already satisfied: pexpect in /usr/local/lib/python3.6/site-packages (4.8.0)
    repo: Requirement already satisfied: ptyprocess>=0.5 in /usr/local/lib/python3.6/site-packages (from pexpect) (0.7.0)
    repo: Collecting ansible
    repo:   Downloading ansible-3.1.0.tar.gz (31.1 MB)
    repo: ERROR: Could not install packages due to an OSError: [Errno 28] No space left on device: '/tmp/pip-install-y4vxk1eb/ansible_ac2c1320a7644857a445e84eeefeb766/ansible_collections/arista/eos/plugins/module_utils/network/eos/rm_templates/ospfv3.py'
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```
I have plenty of disk space available and have set my TMPDIR variable to /home:

```
[root@localhost rhce8env]# df -h
Filesystem      Size  Used Avail Use% Mounted on
devtmpfs         16G     0   16G   0% /dev
tmpfs            16G     0   16G   0% /dev/shm
tmpfs           6.3G  2.1M  6.3G   1% /run
/dev/nvme0n1p3  465G  168G  297G  37% /
/dev/nvme0n1p3  465G  168G  297G  37% /home
/dev/nvme0n1p2  976M  245M  665M  27% /boot
/dev/nvme0n1p1  599M   21M  579M   4% /boot/efi
tmpfs            16G   64K   16G   1% /tmp
tmpfs           3.2G  200K  3.2G   1% /run/user/1000
```
Please let me know if there's any other info I can provide.  Thanks!

### LINKED ISSUE #15: Error: Connection to 127.0.0.1 closed by remote host.
I tried to install the environment by vagrant up but got the below error at stage server1
Note : I am using centos 7 and vmware hypervisor
 
==> server1: Importing base box 'rdbreak/rhel8node'...
==> server1: Matching MAC address for NAT networking...
==> server1: Setting the name of the VM: rhcsa8env_server1_1618497400946_46292
==> server1: Fixed port collision for 22 => 2222. Now on port 2201.
==> server1: Clearing any previously set network interfaces...
==> server1: Preparing network interfaces based on configuration...
    server1: Adapter 1: nat
    server1: Adapter 2: hostonly
==> server1: Forwarding ports...
    server1: 22 (guest) => 2201 (host) (adapter 1)
==> server1: Running 'pre-boot' VM customizations...
==> server1: Booting VM...
==> server1: Waiting for machine to boot. This may take a few minutes...
    server1: SSH address: 127.0.0.1:2201
    server1: SSH username: vagrant
    server1: SSH auth method: private key
==> server1: Machine booted and ready!
==> server1: Checking for guest additions in VM...
    server1: The guest additions on this VM do not match the installed version of
    server1: VirtualBox! In most cases this is fine, but in rare cases it can
    server1: prevent things such as shared folders from working properly. If you see
    server1: shared folder errors, please make sure the guest additions within the
    server1: virtual machine match the version of VirtualBox you have installed on
    server1: your host and reload your VM.
    server1: 
    server1: Guest Additions Version: 5.2.30 r130521
    server1: VirtualBox Version: 6.0
==> server1: Configuring and enabling network interfaces...
==> server1: Rsyncing folder: /root/rhcsa8env/ => /vagrant
==> server1:   - Exclude: [".vagrant/", ".git/"]
There was an error when attempting to rsync a synced folder.
Please inspect the error message below for more info.

Host path: /root/rhcsa8env/
Guest path: /vagrant
Command: "rsync" "--verbose" "--archive" "--delete" "-z" "--copy-links" "--no-owner" "--no-group" "--rsync-path" "sudo rsync" "-e" "ssh -p 2201 -o LogLevel=FATAL   -o ControlMaster=auto -o ControlPath=/tmp/vagrant-rsync-20210415-17981-y72n7j -o ControlPersist=10m  -o IdentitiesOnly=yes -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i '/root/.vagrant.d/boxes/rdbreak-VAGRANTSLASH-rhel8node/1.0/virtualbox/vagrant_private_key'" "--exclude" ".vagrant/" "--exclude" ".git/" "/root/rhcsa8env/" "vagrant@127.0.0.1:/vagrant"
Error: Connection to 127.0.0.1 closed by remote host.
rsync: [sender] write error: Broken pipe (32)
rsync error: unexplained error (code 255) at io.c(820) [sender=3.1.2]

### LINKED ISSUE #17: Repo vm disk size issue
Hi,
after adding epel repo and try the download rpms, the issue occur about disk freespace,

Could you modify disk size in vagrantfile and vdi file for repo vm

Thanks in advance

### LINKED ISSUE #18: Couldn't open file
This is my first time working with vagrant.  I am trying to install on CentOS 8.  I downloaded and installed all of the packages.  I ran vagrant init to initialize.  When I try to start vagrant up, I get 

An error occurred downloading the remote file.

Couldn't open /home/david/base

I can't figure out what is going on.  Can anyone point me in the right direction?  This is all new to me.

### LINKED ISSUE #19: Error 28: no space left on device
RHEL8.4
VirtualBox-6.1-6.1.26_145957_el8-1.x86_64.rpm
vagrant version Installed Version: 2.2.18

repo:   Downloading ansible-4.4.0.tar.gz (35.4 MB)
repo: ERROR: Could not install packages due to an OSError: [Errno 28] No space left on device

### LINKED ISSUE #20: Error 28: no space left on device #19
Solution for the  **Error 28: no space left on device #19**  included into the changes.

_Originally posted by @midu16 in https://github.com/rdbreak/rhce8env/issues/16#issuecomment-902506416_

### PR DESCRIPTION
**Pull request by midu16**
Hello,

I added the "rhel-system-roles-1.0.1-1.el8.noarch" package for the laboratory support. Specially for the requirements that are including the use of the rhel-system-roles. I did not include the install of it, due to the fact the install of ansible and rhel-system-roles are part of exam prerequisites.

### CODE REVIEW

**Comment by midu16** [comment]
Solution for the  **Error 28: no space left on device #19**  included into the changes.

**Review by rdbreak — APPROVED** [review]

**Comment by rdbreak** [comment]
Changes approved. I'll merge the next time I'm back at my computer.

**Comment by midu16** [comment]
thank you for approving it now. I am going to test and clear the appeared conflicts.

**Comment by rdbreak** [comment]
No worries. Thanks for you work on this!

File: Vagrantfile [hunk 1]
```diff
@@ -4,25 +4,31 @@ file_to_disk1 = './disk-0-1.vdi'
...
+      repo.customize ['storageattach', :id,  '--storagectl', 'SATA Controller', '--port', 1, '--device', 0, '--type', 'hdd', '--medium', file_to_disk5]       
+    end      
+  end
+  repo.vm.provision :shell, :inline => "pvs | grep '/dev/sdb' && echo 'The disk was already expanded!' || (pvcreate /dev/sdb; vgextend rhel_rhel8 /dev/sdb; lvextend -l +100%FREE /dev/rhel_rhel8/root; xfs_growfs /dev/rhel_rhel8/root)"
+
   repo.vm.provision :shell, :inline => "sudo sed -i 's/PasswordAuthentication no/PasswordAuthentication yes/g' /etc/ssh/sshd_config; sudo systemctl restart sshd;", run: "always"
   repo.vm.provision :shell, :inline => "yum install https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm -y; sudo yum install -y sshpass python3-pip python3-devel httpd sshpass vsftpd createrepo", run: "always"
   repo.vm.provision :shell, :inline => " python3 -m pip install -U pip ; python3 -m pip install pexpect; python3 -m pip install ansible", run: "always"
-  repo.vm.synced_folder ".", "/vagrant", type: "rsync", rsync__exclude: [".git/", "*.vdi"]
+  repo.vm.synced_folder ".", "/vagrant", type: "rsync", rsync__exclude: ".git/"
```
Comment (McSim85) [inline Vagrantfile:30]: I am not sure, why `vdi` exclusion was removed, but this literally, break rsync process.
So, I would suggest to add that back:
```
repo.vm.synced_folder ".", "/vagrant", type: "rsync", rsync__exclude: [".git/", "*.vdi"]
```

**Pull request merged**